modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpDownloaderTaskExecutor.java

```diff
@@ -128,14 +131,18 @@ public void clusterChanged(ClusterChangedEvent event) {
             // wait for state recovered
             return;
         }
-        // bootstrap downloader after first cluster start
+
+        DiscoveryNode masterNode = event.state().nodes().getMasterNode();
+        if (masterNode == null || masterNode.getVersion().before(Version.V_7_14_0)) {
+            // wait for master to be upgraded so it understands geoip task
+            return;
+        }
+
         clusterService.removeListener(this);
-        if (event.localNodeMaster()) {
```

Comment:
Removing this if check is safe here. Now any node will attempt to start the downloader task if it hasn't started all ready (the start task call is executed on elected master node). This addresses the problem that we was reported, since each time an upgraded node joins the cluster, the geoip downloader task is attempted to be started. So it is unlikely that geoip downloader doesn't exist if cluster has been upgraded.